### ISSUE TITLE
Allow Users to Make Unauthenticated Requests

### ISSUE THREAD

**Issue by thomashzhang** [Sainglend/tda-api-client#4]
Hello, I'm so glad there's a library for this - thank you!

There are plenty of requests for the TD Ameritrade API that don't require authentication (and those are the only ones I use for data gathering). It would be super useful to allow a flag in the `config` object, or even a separate param that's passed into `apiGet` that allows the user to make an unauthenticated request.

Current behavior:

When calling:
```
const getOptionChainResult = await tdaclient.optionchain.getOptionChain(optionsConfig);
```
I get this error:
```
throw new Error('authentication 400: The caller must pass a non null value in the parameter')
```

Desired behavior:
```
const getOptionChainResult = await tdaclient.optionchain.getOptionChain(optionsConfig);
```
I get the options chain back

I've confirmed this works successfully already by commenting out the `getAuthentication` call in the `apiGet` function as a test.

If this project is actively being maintained and I am still working on whatever project I do, I may also be able to contribute a PR

**Comment by Sainglend** [comment]
This capability is already present. In the JSDoc for each such request, it is explained. Below is the one for options. Basically, you should call it with an options object that includes the apiKey property, like this (using the example in samples/codesample.js):

`
const optionsConfig = {
    symbol: 'SPY',
    contractType: tdaclient.optionchain.CONTRACT_TYPE.ALL,
    strikeCount: 60,
    includeQuotes: 'TRUE',
    strategy: tdaclient.optionchain.STRATEGY.SINGLE,
    range: tdaclient.optionchain.RANGE.ALL,
    fromDate: '2020-09-01',
    toDate: '2020-09-10',
    expMonth: tdaclient.optionchain.EXPIRATION_MONTH.SEP,
    optionType: tdaclient.optionchain.OPTION_TYPE.STANDARD,
    apiKey: 'ABCDEFG@AMER.OAUTHAP'
};
const getOptionChainResult = await tdaclient.optionchain.getOptionChain(optionsConfig);
`

JS Doc for getOptionChain:
`
/**
 * Get an options chain for a given symbol. Carefully use config options.
 * Can optionally use apikey for delayed data with an unauthenticated request.
 * @param {Object} config - takes symbol, contractType (ENUM is CONTRACT_TYPE), expMonth (ENUM is EXPIRATION_MONTH), optionType (ENUM is OPTION_TYPE), strategy (ENUM is STRATEGY), range (ENUM is RANGE), includeQuotes, and optionals are:
 * fromDate (yyyy-MM-dd), toDate (yyyy-MM-dd), strikeCount, interval, volatility, underlyingPrice, interestRate, daysToExpiration, apikey
 * @returns {Promise<Object>} api GET result
 * @async
 */
`

**Comment by Sainglend** [comment]
I just pushed up version 1.0.2. I went through all of the functions that use unauthenticated requests, and I noticed two things:
1. Sometimes the url was malformed, i.e. if you were using the apikey it wouldn't append to the url correctly, so it wouldn't be recognized.
2. The authentication refresh was always being done during an http request, regardless of whether the request included the apikey.

So, I'm sorry, you were totally right! These issues should now be fixed.
